packages/backend/src/apps/paysg/__tests__/actions/create-payment.test.ts

```diff
@@ -0,0 +1,101 @@
+import type { IGlobalVariable } from '@plumber/types'
+
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+
+import app from '../..'
+import createPaymentAction from '../../actions/create-payment'
+import getApiBaseUrl from '../../common/get-api-base-url'
+
+const mocks = vi.hoisted(() => ({
+  httpPost: vi.fn(() => ({
+    data: {},
+  })),
+}))
+
+describe('Create payment action', () => {
+  let $: IGlobalVariable
+
+  beforeEach(() => {
+    $ = {
+      auth: {
+        set: vi.fn(),
+        data: {
+          apiKey: 'sample-api-key',
+          paymentServiceId: 'sample-payment-service-id',
+        },
+      },
+      step: {
+        id: 'herp-derp',
+        appKey: 'paysg',
+        position: 2,
+        parameters: {},
+      },
+      http: {
+        post: mocks.httpPost,
+      } as unknown as IGlobalVariable['http'],
+      setActionItem: vi.fn(),
+      app,
+    }
+  })
+
+  afterEach(() => {
+    vi.restoreAllMocks()
+  })
+
+  it('invokes the correct URL based on the API key', async () => {
+    $.auth.data.apiKey = 'paysg_stag_abcd'
+    const expectedBaseUrl = getApiBaseUrl($.auth.data.apiKey)
+
+    await createPaymentAction.run($)
+
+    expect(mocks.httpPost).toHaveBeenCalledWith(
+      '/v1/payment-services/sample-payment-service-id/payments',
+      expect.anything(),
+      expect.objectContaining({
+        baseURL: expectedBaseUrl,
+        headers: {
+          'x-api-key': $.auth.data.apiKey,
+        },
+      }),
+    )
+  })
+
+  it('builds the payload correctly', async () => {
+    $.step.parameters.referenceId = 'test-reference-id'
+    $.step.parameters.payerName = 'test-name'
+    $.step.parameters.payerAddress = 'test-address'
+    $.step.parameters.payerIdentifier = 'test-identifier'
+    $.step.parameters.payerEmail = 'test@email.local'
+    $.step.parameters.description = 'test-description'
+    $.step.parameters.paymentAmountCents = '12345'
+    $.step.parameters.dueDate = '12-12-2023'
+    $.step.parameters.returnUrl = 'test.local'
+    $.step.parameters.metadata = [
```

Comment:
can we have 1 test case with the empty row

Reply:
i add later